Sources/PackageModel/ManifestSourceGeneration.swift

```diff
@@ -100,22 +100,32 @@ fileprivate extension SourceCodeFragment {
     
     /// Instantiates a SourceCodeFragment to represent a single platform.
     init(from platform: PlatformDescription) {
-        // NOTE: This could be cleaned up to use the nicer version accessors.
-        switch platform.platformName {
-        case "macos":
-            self.init(enum: "macOS", string: platform.version)
-        case "maccatalyst":
-            self.init(enum: "macCatalyst", string: platform.version)
-        case "ios":
-            self.init(enum: "iOS", string: platform.version)
-        case "tvos":
-            self.init(enum: "tvOS", string: platform.version)
-        case "watchos":
-            self.init(enum: "watchOS", string: platform.version)
-        case "driverkit":
-            self.init(enum: "DriverKit", string: platform.version)
-        default:
-            self.init(enum: platform.platformName, string: platform.version)
+        // Provides knowledge about what the current version of PackageDescription provides.
+        struct PlatformManifestRepresentation {
```

Comment:
is this information not also found anywhere else? does it make sense to make it more generally available?

Reply:
Unfortunately it isn't.  There is no sharing of state today between what is defined in PackageDescription and what libSwiftPM knows to be defined in PackageDescription.

There are a lot of cases where this would be useful, in particular to share the Codable struct definitions for the stuff that is gets as JSON from the PackageDescription library to SwiftPM proper.  Perhaps the best way would be to be able to share source files between PackageDescription and PackageModel, and have the same types be available in both.  In that case there might be some kind of a runtime CaseIterable thing that we could do here.

Reply:
But to your point, I think perhaps we can figure out something involving a custom source path that reaches into PackageDescription and includes one of its source files where the types are defined.  That would be generally useful and maybe this is a good time to spend that effort rather than add more data like this.

Reply:
side note: today we don't allow overlapping source files between targets (but IIRC it's a restriction we have previously talked about lifting), so we would likely need to use symlinks for now

Reply:
I see — I thought `sources` allowed it and it was just targets directories that couldn't intersect (the `path` parameter).  There seem to be some rather arbitrary rules for what can and cannot be in a target, and I think SwiftPM would do well to relax them.  That's obvious a separate discussion.

I'll see what I can do with a symlink here.